### PR TITLE
chainntnfs/bitcoindnotify: rescan blocks manually instead of rewinding

### PR DESCRIPTION
**Pull request by aakselrod**
Fixes: #1023, #1024, #760.

### CODE REVIEW

File: chainntnfs/bitcoindnotify/bitcoind.go [hunk 1]
```diff
@@ -541,6 +541,7 @@ func (b *BitcoindNotifier) RegisterSpendNtfn(outpoint *wire.OutPoint,
 				return nil, err
 			}
 			b.heightMtx.Lock()
+			defer b.heightMtx.Unlock()
```
[the remark this answers is not in the document]
Comment (cfromknecht) [reply chainntnfs/bitcoindnotify/bitcoind.go]: Gotcha, makes sense

**Comment by chrisrico** [comment]
I'm unable to build, I get the following error:

```# github.com/lightningnetwork/lnd/vendor/github.com/lightninglabs/neutrino
vendor/github.com/lightninglabs/neutrino/blockmanager.go:1233:18: cfheaders.HeaderHashes undefined (type *wire.MsgCFHeaders has no field or method HeaderHashes)
vendor/github.com/lightninglabs/neutrino/blockmanager.go:1247:28: cfheaders.HeaderHashes undefined (type *wire.MsgCFHeaders has no field or method HeaderHashes)
vendor/github.com/lightninglabs/neutrino/blockmanager.go:1281:32: cfhmsg.cfheaders.HeaderHashes undefined (type *wire.MsgCFHeaders has no field or method HeaderHashes)
vendor/github.com/lightninglabs/neutrino/neutrino.go:228:32: not enough arguments in call to wire.NewMsgGetCFHeaders
        have ()
        want (wire.FilterType, uint32, *chainhash.Hash)
vendor/github.com/lightninglabs/neutrino/neutrino.go:229:5: msg.HashStop undefined (type *wire.MsgGetCFHeaders has no field or method HashStop)
vendor/github.com/lightninglabs/neutrino/neutrino.go:231:13: msg.AddBlockLocatorHash undefined (type *wire.MsgGetCFHeaders has no field or method AddBlockLocatorHash)
vendor/github.com/lightninglabs/neutrino/neutrino.go:443:10: msg.HeaderHashes undefined (type *wire.MsgCFHeaders has no field or method HeaderHashes)
vendor/github.com/lightninglabs/neutrino/query.go:390:3: undefined: wire.NewMsgGetCFilter
make: *** [build] Error 2
```

**Comment by aakselrod** [comment]
@chrisrico  I'm going to cherry-pick a commit that fixes this from another PR and then you can `dep ensure` and rebuild. `master` is in a state between merging some `btcd` commits and merging the corresponding `neutrino` commits.

**Comment by aakselrod** [comment]
I lied, the commit is now merged. I'm going to rebase and force-push again.

**Comment by aakselrod** [comment]
@chrisrico Please try again.

**Comment by chrisrico** [comment]
Confirmed fixed.

**Pull request merged**